src/category_theory/filtered.lean

```diff
@@ -112,28 +112,40 @@ instance : is_filtered (discrete punit) :=
 
 namespace is_filtered
 
-variables {C} [is_filtered C]
+section allow_empty
+
+variables {C} [is_filtered_or_empty C]
+
+lemma cocone_objs : ∀ (X Y : C), ∃ Z (f : X ⟶ Z) (g : Y ⟶ Z), true :=
+is_filtered_or_empty.cocone_objs
+lemma cocone_maps : ∀ ⦃X Y : C⦄ (f g : X ⟶ Y), ∃ Z (h : Y ⟶ Z), f ≫ h = g ≫ h :=
+is_filtered_or_empty.cocone_maps
+
+lemma cocone_over_span {i j j' : C} (f : i ⟶ j) (f' : i ⟶ j') :
+  ∃ (k : C) (g : j ⟶ k) (g' : j' ⟶ k), f ≫ g = f' ≫ g' :=
+let ⟨K, G, G', _⟩ := cocone_objs j j', ⟨k, e, he⟩ := cocone_maps (f ≫ G) (f' ≫ G') in
+⟨k, G ≫ e, G' ≫ e, by simpa only [← category.assoc]⟩
```

Comment:
Maybe this one would fit in the `special_shapes` section better? 
`is_cofiltered` has no `special_shapes` section but then probably should have one containing `cone_over_cospan` for symmetry.

Reply:
Good idea; I guess I'll change the name to `span` and `cospan` to match the style of `bowtie` and `tulip` (and golf them as well).

Reply:
I've moved `is_filtered.cocone_over_span` to `special_shapes` but I don't want to create a `special_shapes` section for `is_cofiltered`. Rather, it would be reasonable to remove `max₃` and `coeq₃` (they're no longer used after my golf) and merge `bowtie`/`tulip` into the end of the `allow_empty` section. (These two special shapes were created solely for the purpose of #9101 and we don't have a motivation for the `cofiltered` version.)

Reply:
Seems like a good plan!